Realm/RLMObjectBase.mm

```diff
@@ -209,6 +209,12 @@ + (NSString *)className {
     if ([RLMSwiftSupport isSwiftClassName:className]) {
         className = [RLMSwiftSupport demangleClassName:className];
     }
+    if ([className hasPrefix:@"RLM:"] ) {
```

Comment:
This shouldn't ever actually be hit. During [schema discovery](https://github.com/realm/realm-cocoa/blob/master/Realm/RLMSchema.mm#L132) we [override the className method on each subclass](https://github.com/realm/realm-cocoa/blob/master/Realm/RLMAccessor.mm#L491). That should always happen before any of the subclasses are created, but if this change does something then... I'm not sure. Maybe the unmanaged generated subclass is getting created before schema discovery, and then `RLMReplaceClassNameMethod()` is getting called on it? Maybe `RLMReplaceClassNameMethod()` just plain doesn't work and that somehow doesn't break anything besides this?

Reply:
This is good to know. I’ll investigate further; I know that this change is affecting something so there most likely is a problem lying deeper.

Reply:
I think the cause is the `public final` declarations I added to the definition of `className()` in `Object` for 3.0.0. I'll verify and make a fix tomorrow.